File: apps/cli/src/ui/banner.ts
import pc from "picocolors";

const BANNER_ART = `
  █████╗ ███╗   ███╗ ██████╗ ███████╗██████╗  █████╗ 
 ██╔══██╗████╗ ████║██╔═══██╗██╔════╝██╔══██╗██╔══██╗
 ███████║██╔████╔██║██║   ██║█████╗  ██████╔╝███████║
 ██╔══██║██║╚██╔╝██║██║   ██║██╔══╝  ██╔══██╗██╔══██║
 ██║  ██║██║ ╚═╝ ██║╚██████╔╝███████╗██████╔╝██║  ██║
 ╚═╝  ╚═╝╚═╝     ╚═╝ ╚═════╝ ╚══════╝╚═════╝ ╚═╝  ╚═╝
`;

export function printBanner(version = "0.1.0"): void {
  const lines = BANNER_ART.trim().split("\n");
  const colored = lines.map((line, idx) => {
    // Subtle cyan to blue gradient feel
    if (idx < 2) return pc.cyan(pc.bold(line));
    if (idx < 4) return pc.blue(pc.bold(line));
    return pc.magenta(pc.bold(line));
  }).join("\n");

  console.log(colored);
  console.log(
    `  ${pc.bold(pc.white("Amoeba Framework"))} ${pc.dim(`v${version}`)}  ${pc.dim("—")}  ${pc.italic(pc.cyan("Go Fiber v3 + Modern Frontends"))}\n`
  );
}

File: apps/cli/src/banner.rs
use colored::Colorize;

const BANNER_ART: &str = r#"
  █████╗ ███╗   ███╗ ██████╗ ███████╗██████╗  █████╗ 
 ██╔══██╗████╗ ████║██╔═══██╗██╔════╝██╔══██╗██╔══██╗
 ███████║██╔████╔██║██║   ██║█████╗  ██████╔╝███████║
 ██╔══██║██║╚██╔╝██║██║   ██║██╔══╝  ██╔══██╗██╔══██║
 ██║  ██║██║ ╚═╝ ██║╚██████╔╝███████╗██████╔╝██║  ██║
 ╚═╝  ╚═╝╚═╝     ╚═╝ ╚═════╝ ╚══════╝╚═════╝ ╚═╝  ╚═╝
"#;

pub fn print_banner(version: &str) {
    let lines: Vec<&str> = BANNER_ART.trim().split('\n').collect();
    for (idx, line) in lines.iter().enumerate() {
        if idx < 2 {
            println!("{}", line.cyan().bold());
        } else if idx < 4 {
            println!("{}", line.blue().bold());
        } else {
            println!("{}", line.magenta().bold());
        }
    }
    println!(
        "  {} {} {} {}\n",
        "Amoeba Proteus".white().bold(),
        format!("v{}", version).dimmed(),
        "—".dimmed(),
        "Blazing Fullstack Scaffolding for Go & TypeScript".cyan().italic()
    );
}
